core/src/main/java/io/undertow/server/DefaultByteBufferPool.java

```diff
@@ -153,7 +154,6 @@ public PooledByteBuffer allocate() {
                 local.allocationDepth++;
             }
         }
-        buffer.clear();
```

Comment:
This shouldn't be deleted. E.g. thread local cached buffers will not be cleared anymore.

Reply:
Great catch @ropalka !